build(client): use flub generate api

jason-ha opened the pull request.

Josmithr commented on fluidBuild.config.cjs:
```diff
@@ -77,14 +77,18 @@ module.exports = {
 		"build:test:esm": ["typetests:gen", "build:esnext", "api-extractor:esnext"],
 		"api": {
-			dependsOn: ["api-extractor:commonjs", "api-extractor:esnext"],
+			dependsOn: ["api-extractor:commonjs", "api-extractor:esnext", "api-extractor:report"],
```
If we aren't using API-Extractor to generate the rollups, can't we omit the first two deps here?

jason-ha committed:
build(client,common): use flub generate entrypoints
- replace api-extractor use in api-extractor tasks with flub generate entrypoints
- add build-cli devDep where now needed
- remove then unused api-extractor-cjs.json files
- remove dtsRollup enable from api-extractor configs; now just used for reports

CraigMacomber committed:
Make some public interface members readonly (#20422)

## Description

Remove mutability from a few places in the public API it does not appear
to be needed.

The change to IValueChanged seems like a bug fix since the same object
is provided to all registrants for the event, so they modify the object
it would impact other event receivers which seems like bad coupling. I
Suspect the other changes are similar in nature: I don't think mutation
of any of these was intended to be supported or meaningful, but I could
be wrong about this.

## Breaking Changes

This is a breaking change, since some users of these interfaces could be
modifying them, and this could break.

ChumpChief committed:
Trim exported APIs from odsp-driver and odsp-urlresolver, and strengthen url checks (#20488)

CraigMacomber committed:
tree: Centralize objectNode logic in one file (#20492)

## Description

Continues the pattern started with ArrayNode, and centralize the
ObjectNode logic in one file.

yann-achard-MS committed:
fix(test-dds-utils): minification logic (#20491)

alex-pardes committed:
tree: Add indirection table for node changesets (#20169)

## Description

This refactors ModularChangeset to store node changesets in a table
keyed by a newly introduced node ID. Field specific changesets are no
longer stored recursively, but instead refer to nested changes by node
ID. This is a first step toward refactoring ModularChangeFamily to
enable improving efficiency of its operations.

Sequence changesets no longer need to be templated over the child change
type, but this will be addressed in a subsequent PR.

---------

Co-authored-by: yann-achard-MS <97201204+yann-achard-MS@users.noreply.github.com>
Co-authored-by: Yann Achard <achardy@microsoft.com>

CraigMacomber committed:
tree: Inline ArrayToUnion (#20498)

## Description

Inline ArrayToUnion since its simple and only used once.
This avoids having yet another type computation helper in the public API
surface.

## Breaking Changes

Any external users of ArrayToUnion should replace
`ArrayToUnion<TheArray>` with `TheArray[number]`.

CraigMacomber committed:
Make factory tag object for DataObjectClass readonly (#20500)

## Description

Make factory tag object for DataObjectClass readonly.

CraigMacomber committed:
tree: Centralize mapNode logic in one file (#20501)

## Description

Centralize mapNode logic in one file.

alexvy86 committed:
refactor(tree): Remove unused events in FlexTree (#20503)

seanimam committed:
Adds AppInsightsTelemetryConsumer, new app-insights export subpath and README updates. (#20479)

jenn-le committed:
fix(tree): make commit metadata properties readonly (#20497)

see title

vladsud committed:
Collect better  stats for snapshot loading (SPO) (#20481)

This change
- Collects better telemetry about tree / blob structure of snapshots. In particular, we want to know.
   - number of blobs vs. number of blob leaf nodes in the tree. This difference will be influenced by attachment blobs (missing from the snapshot) and any blob deduping service does. That said, this PR stops short of deduping these cases.
   - Provide better stats on number of trees, include leaf trees. We have "leafTrees" but it was counting all trees, not just leaf nodes.
- Added "fast" parsing case for empty trees
- Added asserts in unit tests that validate our telemetry about slow parsing paths is correct.
- Deleted second version of evalBlobsAndTrees function that was not exported from package and was not used.

yann-achard-MS committed:
fix(tree): enforce integer indices in generic field kind format (#20499)

jason-ha committed:
feat(build-tools): New command `generate:entrypoints` (#20477)

creates API entrypoints per tagged level using simple re-exports from
the given main entrypoint file.

Some special case exports like `export * as foo` (a NamespaceExport) are not handled, but that is also true for api-extractor which is used in concert. exports without recognizable level are ignored which gives correct conservative result but may hinder shaping API as we would really prefer.

Josmithr committed:
refactor(tree): Update `toMapTree` to dispatch on the schema rather than the input data (#20472)

[AB#7546](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7546)

seanimam committed:
makes AppInsightsTelemetryConsumer available under default export pat… (#20506)

## Description
- makes AppInsightsTelemetryConsumer available under default export
paths instead of subpathing,
-  updates API report
- removes extraneous api-report from old package name
"external-telemetry"

Josmithr committed:
build(devtools-view): Update `@fluentui/react-icons` dependency (#20487)

From `^2.0.201` to `^2.0.233`

jason-ha committed:
fix(build-tools): handle special export cases
- Look for namespace exports and source local exports that may have JSDocs that are determined using ExportedDeclarations.

- Change unknown case to be a map and additional data as collected.

- Fix incrementality task typo

jason-ha committed:
docs(build-tools): improved comments
